src/oop/user_defined_classes/hard/q2/MainTest.java

```diff
@@ -0,0 +1,124 @@
+    }
+
+    private static Stream<Arguments> constructorInputProvider() {
+        return Stream.of(
+                Arguments.of(0, 0),
+                Arguments.of(35, 12),
+                Arguments.of(478273, 723662367)
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("constructorInputProvider")
+    public void dishWasherConstructorInitializesAttributesCorrectly(int attribute1, int attribute2) throws Throwable {
+        Object[][] arguments = {
+                {attribute1, int.class},
+                {attribute2, int.class}
+        };
+        Object instance = classInstance.createInstance(arguments);
+        String incorrectAttributeInstantiationMessage =
+                "Your " + testClassName + " constructor does not correctly initialize the %s attribute.";
+        _assertEquals(attribute1, classInstance.getFieldValue(instance, intAttributeName1),
+                String.format(incorrectAttributeInstantiationMessage, intAttributeName1));
+        _assertEquals(attribute2, classInstance.getFieldValue(instance, intAttributeName2),
+                String.format(incorrectAttributeInstantiationMessage, intAttributeName2));
+    }
+
+    private static Stream<Arguments> methodInputProvider() {
+        return Stream.of(
+                Arguments.of(2, 3, 1, 1, 1, 2),
+                Arguments.of(15, 33, 14, 15, 1, 18),
+                Arguments.of(378, 32891, 378, 32891, 0, 0),
+                Arguments.of(0, 0, 0, 0, 0, 0),
+                Arguments.of(5, 10, 0, 1, 5, 9)
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("methodInputProvider")
+    public void correctLoadDishWasherMethod(int attribute1, int attribute2, int remove1, int remove2, int remainder1, int remainder2) throws Throwable {
+        Object[][] arguments = {
+                {attribute1, int.class},
+                {attribute2, int.class}
+        };
+        Object instance = classInstance.createInstance(arguments);
+        assertTrue(classInstance.hasMethod(methodName, new Class[]{int.class, int.class}),
+                "Your " + testClassName + " class is missing the method " + methodName + ". Check that is spelled correctly and has the correct parameters.");
+        assertTrue(classInstance.hasMethod(methodName, new Class[]{int.class, int.class}, Void.TYPE),
+                "Your " + testClassName + " " + methodName + " method does not have the correct return type.");
+        assertTrue(classInstance.hasMethod(methodName, new Class[]{int.class, int.class}, Void.TYPE, new String[]{"public"}),
+                "Your " + testClassName + " " + methodName + " method does not have the correct return type visibility modifier.");
+        classInstance.callMethod(
+                methodName, new Object[][]{{remove1, int.class}, {remove2, int.class}}, null, instance, new Clause[]{
+                        new IntegerLiteral(remainder1),
```

Comment:
Maybe don't test the exact int here, otherwise, if they are not updating the value properly, I think it will just say they are not printing the correct output.

Reply:
You can test that they are printing the correct number after.